src/events.h

```diff
@@ -311,6 +311,8 @@ class DStaticEventHandler : public DObject // make it a part of normal GC proces
 	void WorldThingRevived(AActor* actor);
 	void WorldThingDamaged(AActor* actor, AActor* inflictor, AActor* source, int damage, FName mod, int flags, DAngle angle);
 	void WorldThingDestroyed(AActor* actor);
+	bool WorldHitscanPreFired(AActor* actor, DAngle angle, double distance, DAngle pitch, int damage, FName damageType, PClassActor *pufftype, int flags, double sz, double offsetforward, double offsetside);
+	void WorldHitscanFired(AActor* actor, DVector3 AttackPos, DVector3 DamagePosition, AActor* Inflictor, int flags);
```

Comment:
Make sure to pass vectors by reference value (`const DVector3&`) to avoid recreating them

Reply:
Added